### ISSUE TITLE
Bug: `typescript-estree` does not support `declare namespace foo.bar {}`

### ISSUE THREAD

**Issue by sosukesuzuki** [typescript-eslint/typescript-eslint#7409]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&showAST=es&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCBmA9gQHQBGs8A3vAL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
declare namespace foo.bar {}
```

### Expected Result

```diff
{
  "type": "TSModuleDeclaration",
  "kind": "namespace",
  "body": {
    "type": "TSModuleBlock",
    "body": []
  },
- "declare": false,
+ "declare": true,
  "global": false,
  "id": {
    "left": {
      "decorators": [],
      "name": "astroHTML",
      "optional": false,
      "type": "Identifier"
    },
    "right": {
      "decorators": [],
      "name": "JSX",
      "optional": false,
      "type": "Identifier"
    },
    "type": "TSQualifiedName"
  }
}
```

### Actual Result

```diff
{
  "type": "TSModuleDeclaration",
  "kind": "namespace",
  "body": {
    "type": "TSModuleBlock",
    "body": []
  },
  "declare": false,
  "global": false,
  "id": {
    "left": {
      "decorators": [],
      "name": "astroHTML",
      "optional": false,
      "type": "Identifier"
    },
    "right": {
      "decorators": [],
      "name": "JSX",
      "optional": false,
      "type": "Identifier"
    },
    "type": "TSQualifiedName"
  }
}
```

### Additional Info

context: https://github.com/prettier/prettier/issues/15207

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/rule-tester`       | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `6.2.1` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z`

**Comment by bradzacher** [comment]
interestingly we correctly set the flag for `module`, but not for `namespace`. This also only applies to the nested case, not the single case.

[playground](https://typescript-eslint.io/play/#ts=5.1.6&showAST=es&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCBmA9gfAN7wC%2BAsAFCiSwIoHACuECUARmKRTTXWhwkqDNlzxCBAHSdYvKrXBDGzNh27TQeBTSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by Penguinlay** [comment]
This bug also affects `export namespace`.
